src/webapp/js/actions/index.js

```diff
@@ -85,6 +85,42 @@ export function signoutUser() {
 }
 
 export function resetLogoutCounter() {
```

Comment:
change it to:
export function resetLogoutCounter() {
  return (dispatch) => {
    const token = localStorage.getItem('token');
    const decoded = jwtDecode(token);
    const currentTime = Date.now() / 1000;

    if (decoded.exp < currentTime) {
      const refreshToken = localStorage.getItem('refresh_token');
      dispatch(useRefreshToken(refreshToken));
    }

    dispatch({ type: RESET_LOGOUT_COUNTER });
  };
}

I don't think we need to check if token exists or if the refresh token is still valid, because if they are not we should be logged out before reaching this point, but you can test it to be sure.